### PR TITLE
Add function to get ordered key variables for dataset

### PR DESCRIPTION
**Pull request by nhaydel**
This PR adds a function to get the key variables for a domain. Currently it is a single function to be used in the future by other issues.

Steps to test:
1. Run the unit tests

### CODE REVIEW

File: tests/unit/test_define_xml_reader.py [hunk 1]
```diff
@@ -290,3 +290,12 @@ def test_extract_variable_metadata_not_found():
...
         with pytest.raises(DomainNotFoundInDefineXMLError):
             reader.extract_variables_metadata(domain_name="not found domain")
+
+
+def test_get_domain_key_sequence():
+    with open(test_define_file_path, "rb") as file:
+        contents: bytes = file.read()
+        reader = DefineXMLReaderFactory.from_file_contents(contents)
+        ec_key_sequence: dict = reader.get_domain_key_sequence(domain_name="EC")
+        print(ec_key_sequence)
```
Comment (gerrycampion) [inline tests/unit/test_define_xml_reader.py:300]: remove print

File: cdisc_rules_engine/services/define_xml/base_define_xml_reader.py [hunk 2]
```diff
@@ -320,3 +335,73 @@ def get_define_version(self) -> Optional[str]:
...
         return None
+
+    def get_domain_key_sequence(self, domain_name: str) -> List[str]:
+        """
+        Given a domain name, this function returns the key sequence variable names
+        in order of KeySequence number.
+
+        In the case that Supplemental qualifiers apply to a domain:
+
+        # TODO! How should this be handled - add to end of list
```
Comment (gerrycampion) [inline cdisc_rules_engine/services/define_xml/base_define_xml_reader.py:346]: is this still in question?

File: tests/unit/test_define_xml_reader.py [hunk 3]
```diff
@@ -290,3 +290,12 @@ def test_extract_variable_metadata_not_found():
...
+        assert ec_key_sequence == ["STUDYID", "USUBJID", "ECTRT", "ECSTDTC", "ECREASOC"]
```
Comment (gerrycampion) [inline tests/unit/test_define_xml_reader.py:301]: Nice, it looks like this is doing EC how I would expect. I think you should also add a test for a SUPP dataset as well.
For example, the SUPPEC keys in this define.xml should be:
STUDYID, RDOMAIN, USUBJID, IDVAR, IDVARVAL, QNAM

**Review by gerrycampion — CHANGES_REQUESTED** [review]

File: tests/unit/test_define_xml_reader.py (hunk 3 above)
Comment (nhaydel) [reply tests/unit/test_define_xml_reader.py:301]: Good call, this does need some updating to work with supp datasets

File: cdisc_rules_engine/services/define_xml/base_define_xml_reader.py
Comment (gerrycampion) [inline cdisc_rules_engine/services/define_xml/base_define_xml_reader.py:1]: Sorry, I didn't notice this before, but don't we need to add to `_get_metadata_representation` for this, something like `define_dataset_key_sequence`?
Remember to add to the other define_dataset_ locations, like the builder comments and schema.

Comment (nhaydel) [reply cdisc_rules_engine/services/define_xml/base_define_xml_reader.py:1]: @gerrycampion I assumed that would be added in the future PRs where it is needed

**Review by gerrycampion — APPROVED** [review]

**Pull request merged**